components/director/internal/tenantmapping/mapper_for_system_auth.go

```diff
@@ -95,12 +95,14 @@ func (m *mapperForSystemAuth) getTenantAndScopesForApplicationOrRuntime(sysAuth
 		hasTenant = false
 	}
 
-	if hasTenant && tenant != sysAuth.TenantID {
+	if hasTenant && tenant != *sysAuth.TenantID {
 		return "", "", errors.New("tenant missmatch")
 	}
-
-	tenant = sysAuth.TenantID
-
+	if sysAuth.TenantID != nil {
+		tenant = *sysAuth.TenantID
+	} else {
```

Comment:
You can remove this `else` block as the default for the string type in go is an empty string